src/kivy_state.py

```diff
@@ -0,0 +1,50 @@
+# pylint: disable=too-many-instance-attributes, too-few-public-methods, old-style-class
+
+"""
+Kivy_state.py
+=================================
+"""
+import collections
+
+
+# pylint: disable=old-style-class
+class KivyStateVariables:
+    """This Class hold all the kivy state variables"""
+
+    def __init__(self):
+        self.sqlReady = False  # set to true by sqlTread when ready for processing
+        self.trustedPeer = None
+        self.Peer = collections.namedtuple('Peer', ['host', 'port'])
+        self.association = ''
+        self.navinstance = None
+        self.mail_id = 0
+        self.myAddressObj = None
+        self.detailPageType = None
+        self.ackdata = None
+        self.status = None
+        self.screen_density = None
+        self.msg_counter_objs = None
+        self.check_sent_acc = None
+        self.sent_count = 0
+        self.inbox_count = 0
+        self.trash_count = 0
+        self.draft_count = 0
+        self.all_count = 0
+        self.searcing_text = ''
+        self.search_screen = ''
+        self.send_draft_mail = None
+        self.is_allmail = False
+        self.in_composer = False
+        self.availabe_credit = 0
+        self.in_sent_method = False
+        self.in_search_mode = False
+        self.imageDir = None
+        self.networkProtocolAvailability = None  # Network protocols availability
+        self.resetNetworkProtocolAvailability()
+
+    def resetNetworkProtocolAvailability(self):
+        """This method helps to reset the availability of network protocol"""
+        self.networkProtocolAvailability = {'IPv4': None, 'IPv6': None, 'onion': None}
+
+
+kivy_state_obj = KivyStateVariables()
```

Comment:
instantiation should be in main.py or basically somewhere very early during runtime